src/navigation/ApiSidebar.tsx

```diff
@@ -50,6 +119,14 @@ function useMenuItemHook({ tag, idx }) {
     openVeritcalParent(jQuery(parent.current), mmenuApi);
   };
   const anchor = slug(tag);
+
+  const [currentHash, setCurrentHash] = useState<string>("#");
+  const history: History<any> = useHistory();
+  useEffect(() => {
+    const hash = history.location.hash;
+    setCurrentHash(stripTrailingSlash(hash));
+  }, [history.location.hash]);
```

Comment:
@Locrian24  None of this needs a `useState`. If you want to memoize `currentHash` then just use a `useMemo`

```suggestion
  const [currentHash, setCurrentHash] = useState<string>("#");
  const history: History<any> = useHistory();
  useEffect(() => {
    const hash = history.location.hash;
    setCurrentHash(stripTrailingSlash(hash));
  }, [history.location.hash]);
```